src/Analyzers/CSharp/CodeFixes/UseCollectionExpression/CSharpUseCollectionExpressionForBuilderCodeFixProvider.cs

```diff
@@ -0,0 +1,127 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.UseCollectionInitializer;
+
+namespace Microsoft.CodeAnalysis.CSharp.UseCollectionExpression;
+
+using static CSharpUseCollectionExpressionForBuilderDiagnosticAnalyzer;
+using static SyntaxFactory;
+
+[ExportCodeFixProvider(LanguageNames.CSharp, Name = PredefinedCodeFixProviderNames.UseCollectionExpressionForBuilder), Shared]
+internal partial class CSharpUseCollectionExpressionForBuilderCodeFixProvider
+    : ForkingSyntaxEditorBasedCodeFixProvider<InvocationExpressionSyntax>
+{
+    [ImportingConstructor]
+    [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+    public CSharpUseCollectionExpressionForBuilderCodeFixProvider()
+        : base(CSharpCodeFixesResources.Use_collection_expression,
+               IDEDiagnosticIds.UseCollectionExpressionForBuilderDiagnosticId)
+    {
+    }
+
+    public override ImmutableArray<string> FixableDiagnosticIds { get; } = ImmutableArray.Create(IDEDiagnosticIds.UseCollectionExpressionForBuilderDiagnosticId);
+
+    protected override async Task FixAsync(
+        Document document,
+        SyntaxEditor editor,
+        CodeActionOptionsProvider fallbackOptions,
+        InvocationExpressionSyntax invocationExpression,
+        ImmutableDictionary<string, string?> properties,
+        CancellationToken cancellationToken)
```

Comment:
this logic is similar to what we did for the `XXX.Create(...)` fixer where we construct a new object-creation expression, then convert that to a collection-expression.  But it has added complexity of needing to rewrite lots of statements after teh builder-creation step into collection elements.